### PR TITLE
3059 DOB SSN field space filled

### LINKED ISSUE #3059: Bug: file stuck in pending state when DOB or SSN field is space-filled
Thank you for taking the time to let us know about the issue you found. The basic rule for bug reporting is that 
something isn't working the way one would expect it to work. Please provide us with the information requested 
below and we will look at it as soon as we are able.

## Description

File with space-filled DOB in the TANF T2 record was uploaded and submitted and file status = pending. 

## Action Taken

- uploaded/submitted TANF Active file for FY24 Q1
- reviewed logs and checked submission history. 

## What I expected to see

- file status in a state other than pending
- if errors, an error feedback report generated

## What I did see
file stuck in pending state:
Screenshot 2024-07-02 122823

logs:
```shell=
12:22:48.331: [APP/PROC/WEB.0] 2024-07-02 16:22:48,330 DEBUG util.py::generate_t2_t3_t5_hashes:L277 :  Partial Hash Field Values: T2 202310 11111116935 2 None 999999999
12:22:48.338: [APP/PROC/WEB.0] [2024-07-02 16:22:48,337: ERROR/ForkPoolWorker-1] Task tdpservice.scheduling.parser_task.parse[89ac3089-77cf-4ab1-849c-89b6e7bd79e3] raised unexpected: TypeError('can only concatenate str (not "NoneType") to str')
12:22:48.338: [APP/PROC/WEB.0] Traceback (most recent call last):
12:22:48.338: [APP/PROC/WEB.0] File "/home/vcap/deps/1/python/lib/python3.10/site-packages/celery/app/trace.py", line 451, in trace_task
12:22:48.338: [APP/PROC/WEB.0] R = retval = fun(*args, **kwargs)
12:22:48.338: [APP/PROC/WEB.0] File "/home/vcap/deps/1/python/lib/python3.10/site-packages/celery/app/trace.py", line 734, in __protected_call__
12:22:48.338: [APP/PROC/WEB.0] return self.run(*args, **kwargs)
12:22:48.338: [APP/PROC/WEB.0] File "/home/vcap/app/tdpservice/scheduling/parser_task.py", line 28, in parse
12:22:48.338: [APP/PROC/WEB.0] errors = parse_datafile(data_file, dfs)
12:22:48.338: [APP/PROC/WEB.0] File "/home/vcap/app/tdpservice/parsers/parse.py", line 100, in parse_datafile
12:22:48.338: [APP/PROC/WEB.0] line_errors = parse_datafile_lines(datafile, dfs, program_type, section, is_encrypted, case_consistency_validator)
12:22:48.338: [APP/PROC/WEB.0] File "/home/vcap/app/tdpservice/parsers/parse.py", line 397, in parse_datafile_lines
12:22:48.338: [APP/PROC/WEB.0] should_remove, case_hash_to_remove, case_hash = case_consistency_validator.add_record(record,
12:22:48.338: [APP/PROC/WEB.0] File "/home/vcap/app/tdpservice/parsers/case_consistency_validator.py", line 126, in add_record
12:22:48.338: [APP/PROC/WEB.0] self.duplicate_manager.add_record(record, latest_case_hash, schema, line, line_number)
12:22:48.338: [APP/PROC/WEB.0] File "/home/vcap/app/tdpservice/parsers/duplicate_manager.py", line 188, in add_record
12:22:48.338: [APP/PROC/WEB.0] self.case_duplicate_detectors[case_hash].add_case_member(record, schema, line, line_number)
12:22:48.338: [APP/PROC/WEB.0] File "/home/vcap/app/tdpservice/parsers/duplicate_manager.py", line 138, in add_case_member
12:22:48.338: [APP/PROC/WEB.0] line_hash, partial_hash = schema.generate_hashes_func(line, record)
12:22:48.338: [APP/PROC/WEB.0] File "/home/vcap/app/tdpservice/parsers/util.py", line 279, in generate_t2_t3_t5_hashes
12:22:48.338: [APP/PROC/WEB.0] return hash(line), hash(record.RecordType + str(record.RPT_MONTH_YEAR) + record.CASE_NUMBER +
12:22:48.338: [APP/PROC/WEB.0] TypeError: can only concatenate str (not "NoneType") to str
```

## Other Helpful Information

+ test file: 
[NCtest2.txt](https://github.com/user-attachments/files/16071547/NCtest2.txt)
+ this also happens if the SSN field is space-filled:
```shell=
12:39:39.367: [APP/PROC/WEB.0] 2024-07-02 16:39:39,366 DEBUG fields.py::parse_value:L45 :  Field: 'SSN' at position: [29, 38) is empty.
12:39:39.367: [APP/PROC/WEB.0] 2024-07-02 16:39:39,366 DEBUG fields.py::parse_value:L45 :  Field: 'NEEDS_PREGNANT_WOMAN' at position: [54, 55) is empty.
12:39:39.367: [APP/PROC/WEB.0] 2024-07-02 16:39:39,366 DEBUG fields.py::parse_value:L45 :  Field: 'MONTHS_STATE_TIME_LIMIT' at position: [62, 64) is empty.
12:39:39.367: [APP/PROC/WEB.0] 2024-07-02 16:39:39,366 DEBUG fields.py::parse_value:L45 :  Field: 'JOB_SEARCH_HOP' at position: [84, 86) is empty.
12:39:39.367: [APP/PROC/WEB.0] 2024-07-02 16:39:39,366 DEBUG fields.py::parse_value:L45 :  Field: 'JOB_SEARCH_EA' at position: [86, 88) is empty.
12:39:39.367: [APP/PROC/WEB.0] 2024-07-02 16:39:39,367 DEBUG fields.py::parse_value:L45 :  Field: 'JOB_SEARCH_HOL' at position: [88, 90) is empty.
12:39:39.367: [APP/PROC/WEB.0] 2024-07-02 16:39:39,367 DEBUG fields.py::parse_value:L45 :  Field: 'COMM_SERVICES_HOP' at position: [90, 92) is empty.
12:39:39.367: [APP/PROC/WEB.0] 2024-07-02 16:39:39,367 DEBUG fields.py::parse_value:L45 :  Field: 'COMM_SERVICES_EA' at position: [92, 94) is empty.
12:39:39.367: [APP/PROC/WEB.0] 2024-07-02 16:39:39,367 DEBUG fields.py::parse_value:L45 :  Field: 'COMM_SERVICES_HOL' at position: [94, 96) is empty.
12:39:39.367: [APP/PROC/WEB.0] 2024-07-02 16:39:39,367 DEBUG fields.py::parse_value:L45 :  Field: 'VOCATIONAL_ED_TRAINING_HOP' at position: [96, 98) is empty.
12:39:39.368: [APP/PROC/WEB.0] 2024-07-02 16:39:39,367 DEBUG fields.py::parse_value:L45 :  Field: 'VOCATIONAL_ED_TRAINING_EA' at position: [98, 100) is empty.
12:39:39.368: [APP/PROC/WEB.0] 2024-07-02 16:39:39,367 DEBUG fields.py::parse_value:L45 :  Field: 'VOCATIONAL_ED_TRAINING_HOL' at position: [100, 102) is empty.
12:39:39.368: [APP/PROC/WEB.0] 2024-07-02 16:39:39,367 DEBUG fields.py::parse_value:L45 :  Field: 'JOB_SKILLS_TRAINING_HOP' at position: [102, 104) is empty.
12:39:39.368: [APP/PROC/WEB.0] 2024-07-02 16:39:39,367 DEBUG fields.py::parse_value:L45 :  Field: 'JOB_SKILLS_TRAINING_EA' at position: [104, 106) is empty.
12:39:39.368: [APP/PROC/WEB.0] 2024-07-02 16:39:39,367 DEBUG fields.py::parse_value:L45 :  Field: 'JOB_SKILLS_TRAINING_HOL' at position: [106, 108) is empty.
12:39:39.368: [APP/PROC/WEB.0] 2024-07-02 16:39:39,367 DEBUG fields.py::parse_value:L45 :  Field: 'ED_NO_HIGH_SCHOOL_DIPL_HOP' at position: [108, 110) is empty.
12:39:39.368: [APP/PROC/WEB.0] 2024-07-02 16:39:39,367 DEBUG fields.py::parse_value:L45 :  Field: 'ED_NO_HIGH_SCHOOL_DIPL_EA' at position: [110, 112) is empty.
12:39:39.368: [APP/PROC/WEB.0] 2024-07-02 16:39:39,367 DEBUG fields.py::parse_value:L45 :  Field: 'ED_NO_HIGH_SCHOOL_DIPL_HOL' at position: [112, 114) is empty.
12:39:39.368: [APP/PROC/WEB.0] 2024-07-02 16:39:39,367 DEBUG fields.py::parse_value:L45 :  Field: 'SCHOOL_ATTENDENCE_HOP' at position: [114, 116) is empty.
12:39:39.368: [APP/PROC/WEB.0] 2024-07-02 16:39:39,368 DEBUG fields.py::parse_value:L45 :  Field: 'SCHOOL_ATTENDENCE_EA' at position: [116, 118) is empty.
12:39:39.368: [APP/PROC/WEB.0] 2024-07-02 16:39:39,368 DEBUG fields.py::parse_value:L45 :  Field: 'SCHOOL_ATTENDENCE_HOL' at position: [118, 120) is empty.
12:39:39.368: [APP/PROC/WEB.0] 2024-07-02 16:39:39,368 DEBUG fields.py::parse_value:L45 :  Field: 'PROVIDE_CC_HOP' at position: [120, 122) is empty.
12:39:39.368: [APP/PROC/WEB.0] 2024-07-02 16:39:39,368 DEBUG fields.py::parse_value:L45 :  Field: 'PROVIDE_CC_EA' at position: [122, 124) is empty.
12:39:39.369: [APP/PROC/WEB.0] 2024-07-02 16:39:39,368 DEBUG fields.py::parse_value:L45 :  Field: 'PROVIDE_CC_HOL' at position: [124, 126) is empty.
12:39:39.369: [APP/PROC/WEB.0] 2024-07-02 16:39:39,368 DEBUG fields.py::parse_value:L45 :  Field: 'OTHER_WORK_ACTIVITIES' at position: [126, 128) is empty.
12:39:39.369: [APP/PROC/WEB.0] 2024-07-02 16:39:39,368 DEBUG fields.py::parse_value:L45 :  Field: 'DEEMED_HOURS_FOR_OVERALL' at position: [128, 130) is empty.
12:39:39.369: [APP/PROC/WEB.0] 2024-07-02 16:39:39,368 DEBUG fields.py::parse_value:L45 :  Field: 'DEEMED_HOURS_FOR_TWO_PARENT' at position: [130, 132) is empty.
12:39:39.369: [APP/PROC/WEB.0] 2024-07-02 16:39:39,369 DEBUG validators.py::validate:L791 :  Caught exception in validator: validate__WORK_ELIGIBLE_INDICATOR__HOH__AGE. With field values: {'WORK_ELIGIBLE_INDICATOR': '06', 'RELATIONSHIP_HOH': 1, 'DOB': '99999999'}.
12:39:39.369: [APP/PROC/WEB.0] 2024-07-02 16:39:39,369 DEBUG parse.py::parse_datafile_lines:L387 :  Record #0 from line 4 is invalid.
12:39:39.369: [APP/PROC/WEB.0] 2024-07-02 16:39:39,369 DEBUG util.py::generate_t2_t3_t5_hashes:L277 :  Partial Hash Field Values: T2 202310 11111116935 2 99999999 None
12:39:39.371: [APP/PROC/WEB.0] [2024-07-02 16:39:39,370: ERROR/ForkPoolWorker-1] Task tdpservice.scheduling.parser_task.parse[0c38291e-0c9f-4614-b8cf-4dd575016c96] raised unexpected: TypeError('can only concatenate str (not "NoneType") to str')
12:39:39.371: [APP/PROC/WEB.0] Traceback (most recent call last):
12:39:39.371: [APP/PROC/WEB.0]   File "/home/vcap/deps/1/python/lib/python3.10/site-packages/celery/app/trace.py", line 451, in trace_task
12:39:39.371: [APP/PROC/WEB.0]     R = retval = fun(*args, **kwargs)
12:39:39.371: [APP/PROC/WEB.0]   File "/home/vcap/deps/1/python/lib/python3.10/site-packages/celery/app/trace.py", line 734, in __protected_call__
12:39:39.371: [APP/PROC/WEB.0]     return self.run(*args, **kwargs)
12:39:39.371: [APP/PROC/WEB.0]   File "/home/vcap/app/tdpservice/scheduling/parser_task.py", line 28, in parse
12:39:39.371: [APP/PROC/WEB.0]     errors = parse_datafile(data_file, dfs)
12:39:39.371: [APP/PROC/WEB.0]   File "/home/vcap/app/tdpservice/parsers/parse.py", line 100, in parse_datafile
12:39:39.371: [APP/PROC/WEB.0]     line_errors = parse_datafile_lines(datafile, dfs, program_type, section, is_encrypted, case_consistency_validator)
12:39:39.371: [APP/PROC/WEB.0]   File "/home/vcap/app/tdpservice/parsers/parse.py", line 397, in parse_datafile_lines
12:39:39.371: [APP/PROC/WEB.0]     should_remove, case_hash_to_remove, case_hash = case_consistency_validator.add_record(record,
12:39:39.371: [APP/PROC/WEB.0]   File "/home/vcap/app/tdpservice/parsers/case_consistency_validator.py", line 126, in add_record
12:39:39.371: [APP/PROC/WEB.0]     self.duplicate_manager.add_record(record, latest_case_hash, schema, line, line_number)
12:39:39.371: [APP/PROC/WEB.0]   File "/home/vcap/app/tdpservice/parsers/duplicate_manager.py", line 188, in add_record
12:39:39.371: [APP/PROC/WEB.0]     self.case_duplicate_detectors[case_hash].add_case_member(record, schema, line, line_number)
12:39:39.371: [APP/PROC/WEB.0]   File "/home/vcap/app/tdpservice/parsers/duplicate_manager.py", line 138, in add_case_member
12:39:39.371: [APP/PROC/WEB.0]     line_hash, partial_hash = schema.generate_hashes_func(line, record)
12:39:39.371: [APP/PROC/WEB.0]   File "/home/vcap/app/tdpservice/parsers/util.py", line 279, in generate_t2_t3_t5_hashes
12:39:39.371: [APP/PROC/WEB.0]     return hash(line), hash(record.RecordType + str(record.RPT_MONTH_YEAR) + record.CASE_NUMBER +
12:39:39.371: [APP/PROC/WEB.0] TypeError: can only concatenate str (not "NoneType") to str
1
```
+ URL of the page I was on:
+ Browser and version:
+ Operating System: ( Windows | MacOS X | Linux | Other )
+ Is the issue repeatable?: ( yes | no | don't know )
+ Has the issue occurred more than once?:

### PR DESCRIPTION
**Pull request by raftmsohani**
## Summary of Changes
Pull request closes #3059 _

## How to Test
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d
cd tdrs-backend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d 
```

1. Open http://localhost:3000/ and sign in.
1. Submit the test file in #3059
1. See that it parses

## Deliverables
Space-filled SSN/DOB doesn't stop parsing. Bug is fixed

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [x] Space-filled SSN/DOB doesn't stop parsing

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [x] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [x] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [x] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [x] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

File: tdrs-backend/tdpservice/search_indexes/admin/filters.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def __init__(self, field, request, params, model, model_admin, field_path):
         self.lookup_choices = self._get_lookup_choices(request)
 
     def _get_lookup_choices(self, request):
-        """Filter queryset to guarentee lookup_choices only has STTs associated with the record type."""
+        """Filter queryset to guarantee lookup_choices only has STTs associated with the record type."""
```
Comment (raftmsohani) [inline tdrs-backend/tdpservice/search_indexes/admin/filters.py:54]: This was just a typo

File: tdrs-backend/docker-compose.local.yml [hunk 2]
```diff
@@ -3,7 +3,7 @@ version: "3.4"
 
 services:
   postgres:
-    image: postgres:11.6
+    image: postgres:15.7
```
Comment (raftmsohani) [inline tdrs-backend/docker-compose.local.yml:6]: This is byproduct of postgres upgrade, otherwise tdrs-run-pytest script fails

**Review by elipe17 — APPROVED** [review]
Works as advertised!

**Review by andrew-jameson — APPROVED** [review]

**Review by ADPennington — APPROVED** [review]
ℹ️ i think this PR resolves one of the traceback errors observed in the manual resubmission task @raftmsohani  🥇

testing results below. 
---
[NCtest2.txt -- to test space-filled DOB](https://github.com/user-attachments/files/16383998/NCtest2.txt) 🚀 

- staging
Screenshot 2024-07-25 174553
```shell=
7:43:11.872: [APP/PROC/WEB.0] [2024-07-25 21:43:11,871: ERROR/ForkPoolWorker-1] Task tdpservice.scheduling.parser_task.parse[2d5e48d6-6606-45f1-940c-2cc607a834f9] raised unexpected: TypeError('can only concatenate str (not "NoneType") to str')
17:43:11.872: [APP/PROC/WEB.0] Traceback (most recent call last):
17:43:11.872: [APP/PROC/WEB.0]   File "/home/vcap/deps/1/python/lib/python3.10/site-packages/celery/app/trace.py", line 451, in trace_task
17:43:11.872: [APP/PROC/WEB.0]     R = retval = fun(*args, **kwargs)
17:43:11.872: [APP/PROC/WEB.0]   File "/home/vcap/deps/1/python/lib/python3.10/site-packages/celery/app/trace.py", line 734, in __protected_call__
17:43:11.872: [APP/PROC/WEB.0]     return self.run(*args, **kwargs)
17:43:11.872: [APP/PROC/WEB.0]   File "/home/vcap/app/tdpservice/scheduling/parser_task.py", line 28, in parse
17:43:11.872: [APP/PROC/WEB.0]     errors = parse_datafile(data_file, dfs)
17:43:11.872: [APP/PROC/WEB.0]   File "/home/vcap/app/tdpservice/parsers/parse.py", line 100, in parse_datafile
17:43:11.872: [APP/PROC/WEB.0]     line_errors = parse_datafile_lines(datafile, dfs, program_type, section, is_encrypted, case_consistency_validator)
17:43:11.872: [APP/PROC/WEB.0]   File "/home/vcap/app/tdpservice/parsers/parse.py", line 397, in parse_datafile_lines
17:43:11.872: [APP/PROC/WEB.0]     should_remove, case_hash_to_remove, case_hash = case_consistency_validator.add_record(record,
17:43:11.872: [APP/PROC/WEB.0]   File "/home/vcap/app/tdpservice/parsers/case_consistency_validator.py", line 126, in add_record
17:43:11.872: [APP/PROC/WEB.0]     self.duplicate_manager.add_record(record, latest_case_hash, schema, line, line_number)
17:43:11.872: [APP/PROC/WEB.0]   File "/home/vcap/app/tdpservice/parsers/duplicate_manager.py", line 188, in add_record
17:43:11.872: [APP/PROC/WEB.0]     self.case_duplicate_detectors[case_hash].add_case_member(record, schema, line, line_number)
17:43:11.872: [APP/PROC/WEB.0]   File "/home/vcap/app/tdpservice/parsers/duplicate_manager.py", line 138, in add_case_member
17:43:11.872: [APP/PROC/WEB.0]     line_hash, partial_hash = schema.generate_hashes_func(line, record)
17:43:11.872: [APP/PROC/WEB.0]   File "/home/vcap/app/tdpservice/parsers/util.py", line 273, in generate_t2_t3_t5_hashes
17:43:11.872: [APP/PROC/WEB.0]     return hash(line), hash(record.RecordType + str(record.RPT_MONTH_YEAR) + record.CASE_NUMBER +
17:43:11.872: [APP/PROC/WEB.0] TypeError: can only concatenate str (not "NoneType") to str
```
- qasp


Screenshot 2024-07-25 174615
---
[ME_Active_anon.txt -- to test space-filled SSN](https://github.com/user-attachments/files/16384007/ME_Active_anon.txt)

- staging
Screenshot 2024-07-25 175239

```shell=
17:51:57.211: [APP/PROC/WEB.0] [2024-07-25 21:51:57,210: ERROR/ForkPoolWorker-1] Task tdpservice.scheduling.parser_task.parse[a24eec36-b0ea-473e-a679-b34b4f3c5333] raised unexpected: TypeError('can only concatenate str (not "NoneType") to str')
17:51:57.211: [APP/PROC/WEB.0] Traceback (most recent call last):
17:51:57.211: [APP/PROC/WEB.0]   File "/home/vcap/deps/1/python/lib/python3.10/site-packages/celery/app/trace.py", line 451, in trace_task
17:51:57.211: [APP/PROC/WEB.0]     R = retval = fun(*args, **kwargs)
17:51:57.211: [APP/PROC/WEB.0]   File "/home/vcap/deps/1/python/lib/python3.10/site-packages/celery/app/trace.py", line 734, in __protected_call__
17:51:57.211: [APP/PROC/WEB.0]     return self.run(*args, **kwargs)
17:51:57.211: [APP/PROC/WEB.0]   File "/home/vcap/app/tdpservice/scheduling/parser_task.py", line 28, in parse
17:51:57.211: [APP/PROC/WEB.0]     errors = parse_datafile(data_file, dfs)
17:51:57.211: [APP/PROC/WEB.0]   File "/home/vcap/app/tdpservice/parsers/parse.py", line 100, in parse_datafile
17:51:57.211: [APP/PROC/WEB.0]     line_errors = parse_datafile_lines(datafile, dfs, program_type, section, is_encrypted, case_consistency_validator)
17:51:57.211: [APP/PROC/WEB.0]   File "/home/vcap/app/tdpservice/parsers/parse.py", line 397, in parse_datafile_lines
17:51:57.211: [APP/PROC/WEB.0]     should_remove, case_hash_to_remove, case_hash = case_consistency_validator.add_record(record,
17:51:57.211: [APP/PROC/WEB.0]   File "/home/vcap/app/tdpservice/parsers/case_consistency_validator.py", line 126, in add_record
17:51:57.211: [APP/PROC/WEB.0]     self.duplicate_manager.add_record(record, latest_case_hash, schema, line, line_number)
17:51:57.211: [APP/PROC/WEB.0]   File "/home/vcap/app/tdpservice/parsers/duplicate_manager.py", line 188, in add_record
17:51:57.211: [APP/PROC/WEB.0]     self.case_duplicate_detectors[case_hash].add_case_member(record, schema, line, line_number)
17:51:57.211: [APP/PROC/WEB.0]   File "/home/vcap/app/tdpservice/parsers/duplicate_manager.py", line 138, in add_case_member
17:51:57.211: [APP/PROC/WEB.0]     line_hash, partial_hash = schema.generate_hashes_func(line, record)
17:51:57.211: [APP/PROC/WEB.0]   File "/home/vcap/app/tdpservice/parsers/util.py", line 273, in generate_t2_t3_t5_hashes
17:51:57.211: [APP/PROC/WEB.0]     return hash(line), hash(record.RecordType + str(record.RPT_MONTH_YEAR) + record.CASE_NUMBER +
17:51:57.211: [APP/PROC/WEB.0] TypeError: can only concatenate str (not "NoneType") to str
```

- qasp
Screenshot 2024-07-25 174234

**Pull request merged**